wall_e/src/resources/cogs/ban.py

```diff
@@ -0,0 +1,419 @@
+                              f"**{self.bot.guilds[0].name.upper()}**",
+                              color=discord.Color.red(),
+                              timestamp=datetime.datetime.now(pytz.utc)
+                              )
+        e_obj.add_field(name='Reason',
+                        value=f"```{reason}```\n" +
+                        "**Please refrain from this kind of behaviour in the future. Thank you.**"
+                        )
+
+        e_obj.set_footer(icon_url=self.bot.guilds[0].icon_url, text=self.bot.guilds[0])
+        try:
+            await user.send(embed=e_obj)
+            logger.info("[Ban ban()] User notified via dm of their ban")
+        except (discord.HTTPException, discord.Forbidden, discord.InvalidArgument):
+            dm = False
+            logger.info("[Ban ban()] Notification dm to user failed due to user preferences")
+
+        # kick
+        await user.kick(reason=reason)
+        dt = datetime.datetime.now(pytz.utc)
+        ban.ban_date = dt.timestamp()
+
+        logger.info(f"[Ban ban()] User kicked from guiled at {dt}.")
+
+        # begin purging messages from user in the last purge_window_days
+        await self.purge_messages(ctx, user, purge_window_days)
+
+        # report to council
+        e_obj = discord.Embed(title="Ban Hammer Deployed",
+                              colour=discord.Color.red())
+
+        e_obj.add_field(name="Banned User", value=f"**{ban.username}**", inline=True)
+        e_obj.add_field(name="Moderator", value=f"**{ban.mod}**", inline=True)
+        e_obj.add_field(name="Reason", value=f"```{ban.reason}```", inline=False)
+        e_obj.add_field(name="User Notified via DM", value="*YES*\n" if dm else "*NO*\n*USER HAS DM's DISABLED*",
+                        inline=False)
+        e_obj.set_footer(text="Moderator Action")
+        e_obj.timestamp = dt
+
+        await self.mod_channel.send(embed=e_obj)
+        logger.info(f"[Ban ban()] Message sent to mod channel,{self.mod_channel}, of the ban for {ban.username}.")
+
+        # update database
+        await BanRecords.insert_records(ban)
+
+    async def purge_messages(self, ctx, user: discord.User, timeframe):
+        # first do the ban
+        if timeframe <= 0 or timeframe > 14:
+            await ctx.send('Window to purge message must be between 1 - 14 days. Command aborted')
+
+        # begin purging messages
+        # get list of all channels
+        channels = self.bot.guilds[0].text_channels
+
+        def is_banned_user(msg):
+            msg.author == user
+
+        date = datetime.datetime.now() - datetime.timedelta(timeframe)
+
+        for channel in channels:
+            send_perm = channel.overwrites_for(ctx.guild.default_role).send_messages
+            view_perm = channel.overwrites_for(ctx.guild.default_role).view_channel
+
+            # skip private channels and read-only channels
+            if view_perm is False or send_perm is False:
```

Comment:
I would say to leave a comment here as to why you dont need to purge private or read-only channels.